zio-http/src/main/scala/zio/http/QueryParams.scala

```diff
@@ -88,25 +89,62 @@ final case class QueryParams(map: Map[String, Chunk[String]]) {
    */
   def getAll(key: String): Option[Chunk[String]] = map.get(key)
 
+  /**
+   * Retrieves all typed query parameter values having the specified name.
+   */
+  def getAllAs[A](key: String)(implicit codec: TextCodec[A]): Either[QueryParamsError, Chunk[A]] =
+    map.get(key) match {
+      case Some(params) =>
+        params
+          .map(param => codec.decode(param).toRight(QueryParamsError.MalformedQueryParam(key, param, codec)))
+          .partitionMap(identity) match {
+          case (errors, _) if errors.nonEmpty => Left(QueryParamsError.MultiMalformedQueryParam(errors))
+          case (_, typedParams)               => Right(typedParams)
+        }
+      case None         => Left(QueryParamsError.MissingQueryParam(key))
+    }
+
   /**
    * Retrieves the first query parameter value having the specified name.
    */
   def get(key: String): Option[String] = getAll(key).flatMap(_.headOption)
 
+  /**
+   * Retrieves the first typed query parameter value having the specified name.
+   */
+  def getAs[A](key: String)(implicit codec: TextCodec[A]): Either[QueryParamsError, A] = for {
+    param      <- get(key).toRight(QueryParamsError.MissingQueryParam(key))
```

Comment:
Great! `QueryParamsError.Missing` seems more readable to me. What do you think?

Reply:
Seems logical :)